x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/transform/transforms/latest/LatestDocConfig.java

```diff
@@ -0,0 +1,156 @@
+
+    private static final ParseField UNIQUE_KEY = new ParseField("unique_key");
+    private static final ParseField SORT = new ParseField("sort");
+
+    private final List<String> uniqueKey;
+    private final List<SortBuilder<?>> sort;
+
+    private static final ConstructingObjectParser<LatestDocConfig, Void> STRICT_PARSER = createParser(false);
+    private static final ConstructingObjectParser<LatestDocConfig, Void> LENIENT_PARSER = createParser(true);
+
+    private static ConstructingObjectParser<LatestDocConfig, Void> createParser(boolean lenient) {
+        ConstructingObjectParser<LatestDocConfig, Void> parser =
+            new ConstructingObjectParser<>(
+                NAME, lenient, args -> new LatestDocConfig((List<String>) args[0], (List<SortBuilder<?>>) args[1]));
+
+        parser.declareStringArray(constructorArg(), UNIQUE_KEY);
+        parser.declareField(constructorArg(), (p, c) -> SortBuilder.fromXContent(p), SORT, ObjectParser.ValueType.OBJECT_ARRAY_OR_STRING);
+
+        return parser;
+    }
+
+    public static LatestDocConfig fromXContent(final XContentParser parser, boolean lenient) throws IOException {
+        return lenient ? LENIENT_PARSER.apply(parser, null) : STRICT_PARSER.apply(parser, null);
+    }
+
+    public LatestDocConfig(List<String> uniqueKey, List<SortBuilder<?>> sort) {
+        this.uniqueKey = ExceptionsHelper.requireNonNull(uniqueKey, UNIQUE_KEY.getPreferredName());
+        this.sort = ExceptionsHelper.requireNonNull(sort, SORT.getPreferredName());
+    }
+
+    public LatestDocConfig(StreamInput in) throws IOException {
+        this.uniqueKey = in.readStringList();
+        this.sort = in.readList(FieldSortBuilder::new);
+    }
+
+    public List<String> getUniqueKey() {
+        return uniqueKey;
+    }
+
+    public List<SortBuilder<?>> getSort() {
+        return sort;
+    }
+
+    public ActionRequestValidationException validate(ActionRequestValidationException validationException) {
+        if (uniqueKey.isEmpty()) {
+            validationException = addValidationError("latest_doc.unique_key must be non-empty", validationException);
+        } else {
+            for (int i = 0; i < uniqueKey.size(); ++i) {
+                if (Strings.isNullOrEmpty(uniqueKey.get(i))) {
+                    validationException =
+                        addValidationError("latest_doc.unique_key[" + i + "] element must be non-empty", validationException);
+                }
+            }
+        }
+
+        if (sort.size() != 1) {
+            validationException = addValidationError("latest_doc.sort must have exactly one element", validationException);
+        } else {
+            SortBuilder<?> theOnlySort = sort.get(0);
+            if (theOnlySort instanceof FieldSortBuilder == false) {
+                validationException =
+                    addValidationError(
+                        "latest_doc.sort[0] must be of type FieldSortBuilder, was: " + theOnlySort.getClass().getSimpleName(),
```

Comment:
Is there a chance that an API caller would see this?

If so, we definitely need to adjust this from `FieldSortBuilder` to something API friendly.

Reply:
Done.

The message will be:
```
latest_doc.sort[0] must be sorting based on a document field, was: _script
```

if the caller specifies `_script` sort. Similarly for other sort types.